.travis.yml

```diff
@@ -1,86 +1,83 @@
- - cd ../core
- - if [ "${RUNTEST}" != "0" ]; then cabal configure --enable-tests -v2; else cabal configure -v2; fi
- - cabal build
- - if [ "${RUNTEST}" != "0" ]; then cabal test; fi;
- - cabal check
- - cabal sdist
- - export SRC_TGZ=$(cabal info . | awk '{print $2 ".tar.gz";exit}') ;
-   cd dist/;
-   if [ -f "$SRC_TGZ" ]; then
-      cabal install --force-reinstalls "$SRC_TGZ";
-   else
-      echo "expected '$SRC_TGZ' not found";
-      exit 1;
-   fi
- - cd ../../session
- - cabal configure -v2
- - cabal build
- - cabal check
- - cabal sdist
- - export SRC_TGZ=$(cabal info . | awk '{print $2 ".tar.gz";exit}') ;
-   cd dist/;
-   if [ -f "$SRC_TGZ" ]; then
-      cabal install --force-reinstalls "$SRC_TGZ";
-   else
-      echo "expected '$SRC_TGZ' not found";
-      exit 1;
-   fi
- - cd ../../debug
- - cabal configure -v2
- - cabal build
- - cabal check
- - cabal sdist
- - export SRC_TGZ=$(cabal info . | awk '{print $2 ".tar.gz";exit}') ;
-   cd dist/;
-   if [ -f "$SRC_TGZ" ]; then
-      cabal install --force-reinstalls "$SRC_TGZ";
-   else
-      echo "expected '$SRC_TGZ' not found";
-      exit 1;
-   fi
- - cd ../../
- - make build-openssl-server
- - ghc -threaded -DUSE_CABAL --make test-scripts/TestClient && echo "BUILDING TEST OK" || echo "BUILDING TEST FAILED"
- - if [ -x test-scripts/TestClient ]; then touch debug.log; ./test-scripts/TestClient debug.log; cat debug.log; fi
+- |
+  set -ex
+  if [ "x${RUNTEST}" = "xfalse" ]; then exit 0; fi
+  case "$BUILD" in
+    stack)
+      # create the build stack.yaml
+      case "$RESOLVER" in
+      ghc-7.8)
+        echo "{ resolver: lts-2.22, packages: [ core/, debug/, session/ ], extra-deps: [ psqueues-0.2.5.0, cryptonite-0.24, memory-0.14.14, foundation-0.0.19, basement-0.0.6, x509-1.7.2, x509-system-1.6.6, x509-validation-1.6.9, x509-store-1.6.5, asn1-parse-0.9.4, asn1-types-0.3.2 ], flags: {} }" > stack.yaml
+        stack --no-terminal build --install-ghc --coverage --no-test --no-bench --no-haddock
+        ;;
+      ghc-8.2)
+        echo "{ resolver: lts-10.5, packages: [ core/, debug/, session/ ], extra-deps: [], flags: {} }" > stack.yaml
+        stack --no-terminal build --install-ghc --coverage --test --bench --no-run-benchmarks --haddock --no-haddock-deps
+        ;;
+      ghc-7.10)
+        echo "{ resolver: lts-6.35, packages: [ core/, debug/, session/ ], extra-deps: [ psqueues-0.2.5.0, cryptonite-0.24, memory-0.14.14, foundation-0.0.19, basement-0.0.6, x509-1.7.2, x509-system-1.6.6, x509-validation-1.6.9, x509-store-1.6.5 ], flags: {} }" > stack.yaml
+        stack --no-terminal build --install-ghc --coverage --test --no-bench --haddock --no-haddock-deps
+        ;;
+      ghc-8.0)
+        echo "{ resolver: lts-9.21, packages: [ core/, debug/, session/ ], extra-deps: [ cryptonite-0.24 ], flags: {} }" > stack.yaml
+        stack --no-terminal build --install-ghc --coverage --test --bench --no-run-benchmarks --haddock --no-haddock-deps
+        ;;
+      ghc-8.0)
```

Comment:
This case is duplicated. One should be for Linux but the other should be for macOS. Does this `case` switch for them correctly?

Reply:
no it doesn't, it will go to the first one, but in this case the 2 cases are identical. I need a better way to distinguish OS'es in haskell-ci than the current system